server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -446,11 +469,13 @@ func (s *BuildBuddyServer) UpdateGroup(ctx context.Context, req *grpb.UpdateGrou
 	urlIdentifier := strings.TrimSpace(req.GetUrlIdentifier())
 
 	if urlIdentifier != "" {
-		if group, err = userDB.GetGroupByURLIdentifier(ctx, urlIdentifier); group != nil && group.GroupID != req.GetRequestContext().GetGroupId() {
-			return nil, status.InvalidArgumentError("URL is already in use")
-		} else if err != nil && gstatus.Code(err) != gcodes.NotFound {
+		valid, err := s.validateURLIdentifier(ctx, req, urlIdentifier)
```

Comment:
should we add this validation to `CreateGroup` as well?

also, I wonder if the validation code would fit better in userdb - for better or worse, a lot of the validation logic has ended up there (e.g. the OwnedDomain check), and it's much better tested than this buildbuddy_server package is.

Reply:
good point, done